server/src/main/java/org/elasticsearch/tasks/TaskManager.java

```diff
@@ -292,9 +295,20 @@ public Task unregister(Task task) {
             }
         } finally {
             tracer.stopTrace("task-" + task.getId());
+            for (RemovedTaskListener listener : removedTaskListeners) {
+                threadPool.generic().submit(() -> listener.onRemoved(task));
```

Comment:
Do we need to fork for every listener? I think everything the listener does is now lightweight except for completing the final listener when all tasks are removed. We should just fork at the end there instead.